src/Raven.Server/NotificationCenter/Indexing.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Util;
+using Raven.Server.NotificationCenter.Notifications;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Sparrow.Json;
+using Sparrow.Logging;
+
+namespace Raven.Server.NotificationCenter
+{
+    public class Indexing : IDisposable
+    {
+        private static readonly string Source = "Indexing";
+        private static readonly string IndexingTableId = PerformanceHint.GetKey(PerformanceHintType.Indexing, Source);
+
+        private readonly NotificationCenter _notificationCenter;
+        private readonly NotificationsStorage _notificationsStorage;
+        private readonly string _database;
+
+        private DateTime _warningUpdateTime = new DateTime();
+        private ConcurrentQueue<(string Action, WarnIndexOutputsPerDocument.WarningDetails warningDetails)> _warningQueue = 
+            new ConcurrentQueue<(string Action, WarnIndexOutputsPerDocument.WarningDetails warningDetails)>();
+        
+        private Timer _indexingTimer;
+        private readonly Logger _logger;
+        private readonly object _locker = new object();
+
+        public Indexing(NotificationCenter notificationCenter, NotificationsStorage notificationsStorage, string database)
+        {
+            _notificationCenter = notificationCenter;
+            _notificationsStorage = notificationsStorage;
+            _database = database;
+            _logger = LoggingSource.Instance.GetLogger(database, GetType().FullName);
+        }
+
+        public void AddWarning(string indexName, WarnIndexOutputsPerDocument.WarningDetails _indexOutputsWarning)
+        {
+            var now = SystemTime.UtcNow;
+            var update = _warningUpdateTime;
+
+            if (now - update < TimeSpan.FromSeconds(15))
+                return;
+
+            _warningUpdateTime = now;
+            
+            _indexOutputsWarning.LastWarningTime = now;
+            _warningQueue.Enqueue((indexName, _indexOutputsWarning));
```

Comment:
you put that in queue: new ConcurrentQueue<(string Action, WarnIndexOutputsPerDocument.WarningDetails warningDetails)>();

should rename paramter name Action -> IndexName?

Reply:
done